### PR TITLE
feat: dpi responsive layout

### PR DESCRIPTION
**Pull request by BrentBoyMeBob**
# Details
This addresses issue #333 and the Reddit post [here](https://www.reddit.com/r/revancedapp/comments/xpdilc/larger_font_support_needed_im_unable_to_use_a/), fixing the issue of widgets overflowing on low resolution or very high DPI settings. It adds flexibility to a chunk of widgets available, and integrates flutter_screenutil to scale the text when needed.

### CODE REVIEW

**Comment by BrentBoyMeBob** [comment]
(edit: the wrap isn't spacing, I'll fix that real quick)

**Comment by Aunali321** [comment]
I will take a look at this later.

**Comment by Aunali321** [comment]
Screenshot_20221010-001640
Some ui elements are too small like button here. Please make sure UI elements are the size before and after your changes when tested on the same device.

**Comment by BrentBoyMeBob** [comment]
That makes sense; I'll adjust that real quick.

**Comment by Aunali321** [comment]
@afnzmn can you test this and make sure the ui is same as our design?

**Comment by BrentBoyMeBob** [comment]
Any updates on testing by chance?

**Comment by Aunali321** [comment]
@Ushie is this PR ready to be merged?

**Comment by Ushie** [comment]
> @Ushie is this PR ready to be merged?

Yes and no, it doesn't pass the requirement you have if looking exactly the same, some values could probably be tinkered to achieve that like increase font size a bit, nevertheless I don't have the time for this PR and nobody seems to work on it, it **works** and if we can compromise on the exact look it's a-okay, just need to sort the merge conflict

**Comment by BrentBoyMeBob** [comment]
Just resolved some conflicts, which caused some errors in the scripts, I'll quickly fix those.

**Comment by 2shrestha22** [comment]
`flutter_screenutil` will make UI look bigger in bigger devices and smaller in smaller devices. If device is very small fonts may become unreadable. We should try to make app responsive by updating its layout not by changing the widget size.

**Comment by BrentBoyMeBob** [comment]
My opinion personally, I actually think the opposite. Padding and widget size isn't really being affected, moreso the size of the text itself, but at the moment, the text is too large on those smaller displays, even with layout optimizations (which I did indeed apply some), everything gets cramped quickly, and due to lack of optimization, results in a ton of overflow errors. The phone that's used in question is the S10e, which isn't even that small of a device.

**Comment by BrentBoyMeBob** [comment]
Finished the said commit. I have tweaked some of the values (along with simplifying some operations, redoing layout optimizations, and bringing it up to speed from the old conflicts), so it should look closer to how one would expect.

**Comment by ILoveOpenSourceApplications** [comment]
Is there any issue with this commit holding it from getting merged?

**Comment by BrentBoyMeBob** [comment]
As far as I'm aware, not really, I don't know what's stopping it at this point.

If it is to be merged, I'll need to bring it back up to speed by resolving the conflicts (which I can definitely do), but first, I want to make sure that the maintainers are going through with this, since this PR doesn't seem to be of priority.

**Comment by Ushie** [comment]
I'll take a second look when you resolve the conflicts, could you try making it look as close as the existing ReVanced Manager though? if I remember correctly, text sizes were decently smaller last time I checked this out

**Comment by ILoveOpenSourceApplications** [comment]
@BrentBoyMeBob can we have the conflicts resolved and then request for another review?

File: lib/ui/views/settings/settings_view.dart
Comment (dhruvanbhalara) [inline lib/ui/views/settings/settings_view.dart:27]: I suggest do not initialize the `screen_util` this way, adding it into the widget tree and use it wherever it's possible while adding into the widget tree from the top-level pass the design size based on that it'll adapt responsive font size.

File: lib/ui/widgets/patcherView/app_selector_card.dart
Comment (dhruvanbhalara) [inline lib/ui/widgets/patcherView/app_selector_card.dart:55]: Newer `screen_util` allows you to make adaptive spaces between widgets, you can use it here.

**Review by dhruvanbhalara — CHANGES_REQUESTED** [review]

File: lib/ui/views/settings/settings_view.dart
Comment (BrentBoyMeBob) [reply lib/ui/views/settings/settings_view.dart:27]: Fair point, definitely an oversight I did.

File: lib/ui/widgets/patcherView/app_selector_card.dart
Comment (BrentBoyMeBob) [reply lib/ui/widgets/patcherView/app_selector_card.dart:55]: Good to know! This could definitely help with UI consistency, as sometimes, the padding doesn't match.

**Comment by BrentBoyMeBob** [comment]
The old code doesn't quite fit with the current codebase anymore, and had some glaring issues, so I'm currently going to redo it. Will update when I'm committing again.

**Comment by Ushie** [comment]
Sounds good

**Comment by BrentBoyMeBob** [comment]
There are still a lot of overflow issues, but I don't know if flutter_screenutil is even necessary after re-evaluating, especially since there have been some changes in cleanliness in layout since then, and font size isn't really the issue after inspection, it's how the text is handled in relation to the scale that causes them. Plus, it's much easier to manage since font size is effectively unaffected, so it still looks visually the same on lower DPI devices.

To be honest, a lot of these can be solved with fixes that are conscious to screen size, such as adapting the often noted "Patched Applications" section to some tweaks with LayoutBuilder.

**Comment by BrentBoyMeBob** [comment]
@Ushie How does this look? I decided to go through with removing flutter_screenutil entirely, and now it mainly readjusts the layout depending on the room given in widgets, which should mean that devices with conventional screen sizes and DPI should look the same as before, whilst making the needed changes to more zoomed in ones.

**Comment by Ushie** [comment]
@TheAabedKhan Could you please take a look at this PR? it achieves better responsive UI which has been an issue in ReVanced Manager for smaller-sized devices where components end up going out of their containers and unusable

File: lib/ui/widgets/patcherView/app_selector_card.dart
Comment (TheAabedKhan) [inline lib/ui/widgets/patcherView/app_selector_card.dart:60]:
```suggestion
                    Flexible(
                      child: Text(
                        locator<PatcherViewModel>().getAppSelectionString(),
                        style: const TextStyle(fontWeight: FontWeight.w600),
                      ),
                    ),
```
This is it. No LayoutBuilder is needed.

File: lib/ui/widgets/shared/application_item.dart
Comment (TheAabedKhan) [inline lib/ui/widgets/shared/application_item.dart:79]:
```suggestion
          header: Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: <Widget>[
              Flexible(
                child: Row(
                  children: [
                    SizedBox(
                      width: 40,
                      child: Image.memory(widget.icon, height: 40, width: 40),
                    ),
                    const SizedBox(width: 19),
                    Expanded(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: <Widget>[
                          Text(
                            widget.name,
                            maxLines: 1,
                            overflow: TextOverflow.ellipsis,
                            style: const TextStyle(
                              fontSize: 16,
                              fontWeight: FontWeight.w500,
                            ),
                          ),
                          Text(
                            format(widget.patchDate),
                            maxLines: 1,
                            overflow: TextOverflow.ellipsis,
                            style: const TextStyle(
                              fontSize: 16,
                              fontWeight: FontWeight.w500,
                            ),
                          ),
                        ],
                      ),
                    ),
                  ],
                ),
              ),
              Row(
                children: [
                  RotationTransition(
                    turns: Tween(begin: 0.0, end: 0.50)
                        .animate(_animationController),
                    child: const Padding(
                      padding: EdgeInsets.all(8.0),
                      child: Icon(Icons.arrow_drop_down),
                    ),
                  ),
                  const SizedBox(width: 8),
                  Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    crossAxisAlignment: CrossAxisAlignment.end,
                    children: <Widget>[
                      CustomMaterialButton(
                        label: widget.isUpdatableApp
                            ? I18nText('applicationItem.patchButton')
                            : I18nText('applicationItem.infoButton'),
                        onPressed: widget.onPressed,
                      ),
                    ],
                  ),
                ],
              ),
            ],
          ),
          collapsed: const SizedBox(),
          expanded: Padding(
```
How about this?

**Review by TheAabedKhan — CHANGES_REQUESTED** [review]

**Comment by BrentBoyMeBob** [comment]
@TheAabedKhan All of your suggestions have worked perfectly it seems, including the larger application_item block, so I've applied them in the newest commit.

File: lib/ui/widgets/patcherView/app_selector_card.dart
Comment (TheAabedKhan) [inline lib/ui/widgets/patcherView/app_selector_card.dart:41]: Remove the `LayoutBuilder` as it is not required now.

**Review by TheAabedKhan — CHANGES_REQUESTED** [review]

File: lib/ui/widgets/patcherView/app_selector_card.dart
Comment (BrentBoyMeBob) [reply lib/ui/widgets/patcherView/app_selector_card.dart]: @TheAabedKhan Just did that, anything else needed?

**Review by TheAabedKhan — APPROVED** [review]
All good ✨

**Review by validcube — APPROVED** [review]
🎉 Tested, works great!

<details>
<summary>💫 Preview</summary>

image
image

</details>

**Comment by validcube** [comment]
@Ushie Any final suggestions before merge?

**Comment by Ushie** [comment]
Did you test all pages?

- App selector
- Patches selector
- Installer
- Settings
- Dialogs in settings

**Comment by ILoveOpenSourceApplications** [comment]
> Did you test all pages?
> 
> * App selector
> * Patches selector
> * Installer
> * Settings
> * Dialogs in settings

@validcube  have you tested the above mentioned pages?

**Comment by validcube** [comment]
> > Did you test all pages?
> > 
> > * App selector
> > * Patches selector
> > * Installer
> > * Settings
> > * Dialogs in settings
> 
> @validcube  have you tested the above mentioned pages?

Planning to test it today, I'm not at home at the moment

**Comment by Ushie** [comment]
Low and behold, the long awaited merge, thank you lots @BrentBoyMeBob for your patience and I'm glad to see your PR finally ship after such a long time in review, sorry about that!

**Pull request merged**
